src/pages/Homepage.jsx

```diff
@@ -0,0 +1,23 @@
+import React, { useState } from 'react';
+import primaryBooks from '../components/Book';
+import Booklist from '../components/Booklist';
+import Form from '../components/Form';
+
+const Homepage = () => {
+  const [books, setBooks] = useState(primaryBooks);
+
+  const handleAddBook = (newBook) => {
+    // Use the functional form of setBooks
+    setBooks((prevBooks) => [...prevBooks, newBook]);
+    console.log('Books:', books);
```

Comment:
- [ ] It will be great to remove the `console.log` statement. This is because it kills application performance and also displays sensitive information about your application to the console.